index.js

```diff
@@ -138,6 +138,32 @@ function getSequelizeTypeFromJoi(dialect, type, rules) {
     }
 }
 
+/**
+ * Checks if default value is allowed for the specified dialect and Sequelize type
+ * @method isDefaultValueAllowed
+ * @param  {String}          dialect   Underlying system that we're writing to
+ * @param  {String}          fieldType Sequelize type
+ * @return {Boolean}                   true if default value is allowed. false otherwise.
+ */
+function isDefaultValueAllowed(dialect, fieldType) {
+    // Few models have defined default values for types that do not allow default values.
+    // Ex: pipeline.parameters https://github.com/screwdriver-cd/data-schema/blob/master/models/pipeline.js#L86
+    // To keep the DB sync backward compatible, we need to skip setting default values for such types.
+    // For more details, refer https://github.com/sequelize/sequelize/blob/0016739f2a4af5215ebbbe3724f8298c7335ed88/src/dialects/mysql/query-generator.js#L27
+    // TODO: Remove this when the models are refactored to not define default values for such types.
+    if (dialect === 'mysql') {
+        const typesWithoutDefault = [Sequelize.BLOB, Sequelize.TEXT, Sequelize.GEOMETRY, Sequelize.JSON];
```

Comment:
is TEXT same as MEDIUMTEXT ?

Reply:
`MEDIUMTEXT`  is a variation of `TEXT` .  If length is specified Ex: "medium" , `toSql()` invoked by `toString()` in `SEQUILIZE.TEXT` translates to `"MEDIUMTEXT"`  type for the underlying database